src/v/config/configuration.h

```diff
@@ -88,16 +88,18 @@ struct configuration final : public config_store {
     property<std::chrono::milliseconds> metadata_dissemination_retry_delay_ms;
     property<int16_t> metadata_dissemination_retries;
     property<std::chrono::milliseconds> tm_sync_timeout_ms;
-    property<model::violation_recovery_policy> tm_violation_recovery_policy;
+    enum_property<model::violation_recovery_policy>
+      tm_violation_recovery_policy;
     property<std::chrono::milliseconds> rm_sync_timeout_ms;
     property<uint32_t> seq_table_min_size;
     property<std::chrono::milliseconds> tx_timeout_delay_ms;
-    property<model::violation_recovery_policy> rm_violation_recovery_policy;
+    enum_property<model::violation_recovery_policy>
+      rm_violation_recovery_policy;
     property<std::chrono::milliseconds> fetch_reads_debounce_timeout;
     property<std::chrono::milliseconds> alter_topic_cfg_timeout_ms;
     property<model::cleanup_policy_bitflags> log_cleanup_policy;
```

Comment:
`cleanup_policy_bitflags` is also an `enum`, but with specified values. I don't see any reason why we can't use `enum_property` for it also

Reply:
It's not quite an enum: for that property one can specify a union of values like `compact,delete`, whereas enum_property expects a single value to be chosen.